src/test/scala/chisel3/tests/AsyncResetNativeTest.scala

```diff
@@ -0,0 +1,171 @@
+// See LICENSE for license details.
+
+package chisel3.tests
+
+import chisel3._
+import chisel3.experimental.MultiIOModule
+import chisel3.tester._
+import chisel3.tester.experimental.TestOptionBuilder._
+import chisel3.tester.internal.VerilatorBackendAnnotation
+import firrtl.AnnotationSeq
+import org.scalatest._
+
+/**
+  * circuit that illustrates usage of async register
+  * @param resetValue value on reset
+  */
+class NativeAsyncResetRegModule(resetValue: Int) extends MultiIOModule {
+  //noinspection TypeAnnotation
+
+  val io = IO(new Bundle {
+    val in = Input(UInt(8.W))
+    val out = Output(UInt(8.W))
+  })
+  val asyncReset = reset.asAsyncReset
+
+  val reg = withClockAndReset(clock, asyncReset) {
+    RegInit(resetValue.U(8.W))
+  }
+
+  reg := io.in
+  io.out := reg
+}
+
+class NativeAsyncResetFeedbackModule() extends MultiIOModule {
```

Comment:
Same comments as above: what's the noinspect TypeAnnotation, prefer not needing the Native prefix, inlining the asyncReset, and moving the assign / updates to inside the reset scope.